src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java

```diff
@@ -260,6 +217,16 @@ public TopicPartitionChannel(
             && this.conn != null
             && this.conn.hasSchemaEvolutionPermission(
                 tableName, sfConnectorConfig.get(SNOWFLAKE_ROLE));
+
+    // Open channel and reset the offset in kafka
+    this.channel = Preconditions.checkNotNull(openChannelForTable());
+    final long lastCommittedOffsetToken = fetchOffsetTokenWithRetry();
+    this.offsetPersistedInSnowflake.set(lastCommittedOffsetToken);
+    this.processedOffset.set(lastCommittedOffsetToken);
+    if (lastCommittedOffsetToken != NO_OFFSET_TOKEN_REGISTERED_IN_SNOWFLAKE) {
+      // Reset the offset in kafka, only if we have a valid offset token at server side
+      this.sinkTaskContext.offset(this.topicPartition, lastCommittedOffsetToken);
+    }
```

Comment:
This sounds like a great idea. Do you mind adding a comment in the constructor definition? 
Technically this is happening even before we receive any offsets from put API. It would be great value addition to our javadocs.